src/schema.ts

```diff
@@ -198,6 +198,35 @@ export class Schema {
     }
   }
 
+  /**
+   * Given an EncodingQuery with a timeUnit, returns true if the number of
+   * dates that are completely distinct in all parts of the timeUnit is greater than 1.
+   * i.e.
+   * ('yearmonth', [Jan 1 2000, Feb 2 2000] returns false)
+   * ('yearmonth', [Jan 1 2000, Feb 2 2001] returns true)
+   */
+  public timeUnitHasVariation(encQ: EncodingQuery): boolean {
+    // if there is no variation in `date`, there should not be variation in `day`
+    if (encQ.timeUnit === TimeUnit.DAY) {
+      const dateEncQ: EncodingQuery = extend({}, encQ);
+      dateEncQ.timeUnit = TimeUnit.DATE;
```

Comment:
you shouldn't modify a `const`  ... put `{timeUnit: ...}` as the last param in the `extend`?